src/main/kotlin/uk/gov/dluhc/printapi/messaging/ApplicationRemovedMessageListener.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.dluhc.printapi.messaging
+
+import io.awspring.cloud.messaging.listener.annotation.SqsListener
+import mu.KotlinLogging
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.messaging.models.ApplicationRemovedMessage
+import uk.gov.dluhc.printapi.service.CertificateDataRetentionService
+import javax.validation.Valid
+
+private val logger = KotlinLogging.logger { }
+
+/**
+ * Implementation of [MessageListener] to handle [ApplicationRemovedMessage] messages
+ */
+@Component
+class ApplicationRemovedMessageListener(
+    val certificateDataRetentionService: CertificateDataRetentionService
+) : MessageListener<ApplicationRemovedMessage> {
+
+    @SqsListener("\${sqs.application-removed-queue-name}")
+    override fun handleMessage(@Valid @Payload payload: ApplicationRemovedMessage) {
```

Comment:
Just so I'm clear .... a `ApplicationRemovedMessage` is for a VAC Application? And this listener can also be used to handle doing the data retention for a VAC Application's Temporary Certificate(s) as well as and when we get there?

Does that mean we are proposing other message types in the future, such as `AnonymousApplicationRemovedMessage`, `PostalApplicationRemovedMessage`, `OverseasApplicationRemovedMessage` etc? (not that those last 2 have 'certificates' I dont think, but you get the idea)

Or, does `ApplicationRemovedMessage` have a field in it to say whether its a regular VAC or an Anonymous application ?

Reply:
I considered including a type field, but I don't think there's any data that needs to be removed after the initial retention period for temp certs or AEDs (since they have no address). I didn't want to overthink it, so decided I could always add a type later if necessary.

Reply:
Thats fair. Thanks 👍